example/streamed_list_objects/main.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	openfga "github.com/openfga/go-sdk"
+	"github.com/openfga/go-sdk/client"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Get API URL from environment or use default
+	apiUrl := os.Getenv("FGA_API_URL")
+	if apiUrl == "" {
+		apiUrl = "http://localhost:8080"
+	}
+
+	// Create initial client for store creation
+	config := client.ClientConfiguration{
+		ApiUrl: apiUrl,
+	}
+	fgaClient, err := client.NewSdkClient(&config)
+	if err != nil {
+		handleError(err)
+		return
+	}
+
+	fmt.Println("Creating temporary store")
+	store, err := fgaClient.CreateStore(ctx).Body(client.ClientCreateStoreRequest{
+		Name: "streamed-list-objects",
+	}).Execute()
+	if err != nil {
+		handleError(err)
+		return
+	}
+
+	// Create client with store ID
+	clientWithStore, err := client.NewSdkClient(&client.ClientConfiguration{
+		ApiUrl:  apiUrl,
+		StoreId: store.Id,
+	})
+	if err != nil {
+		handleError(err)
+		return
+	}
+
+	fmt.Println("Writing authorization model")
+	authModel, err := writeAuthorizationModel(ctx, clientWithStore)
+	if err != nil {
+		handleError(err)
+		return
+	}
+
+	// Create final client with store ID and authorization model ID
+	fga, err := client.NewSdkClient(&client.ClientConfiguration{
+		ApiUrl:               apiUrl,
+		StoreId:              store.Id,
+		AuthorizationModelId: authModel.AuthorizationModelId,
+	})
+	if err != nil {
+		handleError(err)
+		return
+	}
+
+	fmt.Println("Writing tuples (1000 as owner, 1000 as viewer)")
+	if err := writeTuples(ctx, fga); err != nil {
+		handleError(err)
+		return
+	}
+
+	fmt.Println("Streaming objects via computed 'can_read' relation...")
+	if err := streamObjects(ctx, fga); err != nil {
+		handleError(err)
+		return
+	}
+
+	fmt.Println("Cleaning up...")
+	if _, err := fga.DeleteStore(ctx).Execute(); err != nil {
+		fmt.Printf("Failed to delete store: %v\n", err)
+	}
+
+	fmt.Println("Done")
+}
+
+func writeAuthorizationModel(ctx context.Context, fgaClient *client.OpenFgaClient) (*client.ClientWriteAuthorizationModelResponse, error) {
+	// Define the authorization model with computed relations
+	ownerUserset := openfga.Userset{This: &map[string]interface{}{}}
```

Comment:
Maybe similar to @daniel-jonathan's comment on Java, we could choose to use [language](https://github.com/openfga/language/blob/main/pkg/go/README.md) here if we wanted to use DSL over the JSON model

Reply:
Oh yes we could do that. Let me refactor it